src/libraries/System.Net.Primitives/tests/UnitTests/CookieExceptionTest.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Net.Primitives.UnitTests.Tests;
+
+public sealed class CookieExceptionTest
+{
+    [Theory]
+    [InlineData("Testing the CookieException")]
+    [InlineData(null)]
+    public void Constructor_Message(string? message)
+    {
+        var exception = new CookieException(message);
+        Assert.Equal(message, exception.Message);
+    }
+
+    [Theory]
+    [InlineData("Testing the CookieException", true)]
+    [InlineData("Testing the CookieException", false)]
+    [InlineData(null, true)]
+    [InlineData(null, false)]
+    public void Constructor_Message_InnerException(string? message, bool innerException)
+    {
+        var inner = innerException ? new Exception() : null;
+        var exception = new CookieException(message, inner);
+
+        Assert.Equal(message, exception.Message);
+        Assert.Equal(inner, exception.InnerException);
```

Comment:
```suggestion
    [Fact]
    public void Constructor_Message()
    {
        var exception = new CookieException("Foo");
        Assert.Equal("Foo", exception.Message);
    }

    [Theory]
    [InlineData(true)]
    [InlineData(false)]
    public void Constructor_Message_InnerException(bool innerException)
    {
        var inner = innerException ? new Exception() : null;
        var exception = new CookieException("Foo", inner);

        Assert.Equal("Foo", exception.Message);
        Assert.Equal(inner, exception.InnerException);
```

The tests are failing because `CookieException` derives from `FormatException`, which sets a message if you give it a `null`.